### ISSUE TITLE
Can I define a one to many between two views and using an expression of dates between

### ISSUE THREAD

**Issue by tannenbaums** [dotnet/efcore#25190]
## Ask a question

Hi,

I believe I have a pretty unique requirement here, and I'm trying to get it working with ef core so far to no avail.
I have two views that both must be a view because they are both a calculation between two tables.
The relationship between them is one-to-many but not in the conventional way, meaning not just using Id's, but rather a combination of Id's and an expression of StartDate and EndDate of the sub-view falling within range of the Main view.

something like this:

### Include your code

View1
Id
StartDate
EndDate
Some more info
List&lt;View2&gt; Notes {get;set;}

View2
View1Id //this will capture multiple rows in view1 - thus we need to also add a date expression
StartDate
EndDate
Note
View1 View1Prop{get;set;}

I'm looking for a way to do 

view2.HasOne(v2 => v2.View1)
                    .WithMany(v1 => v1.Notes)
                    .HasForeignKey(v2 => v2.View1Id && v1.startDate >= v2.startDate && v1.EndDate <= v2.EndDate)
     
In other words, the id's are not enough to link the two. It's a combination of Id plus the dates expression that will capture what rows in view2 needs to be in the Notes Collection in View1. Is it possible using EF core?              .
Thank you so much in advance.
### Include provider and version information

EF Core version:
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: NET 5.0
Operating system: Win 2019 server
IDE: Visual Studio 2019 16.10.2

**Comment by tannenbaums** [comment]
Never mind, i figured it out. For anyone out there facing the same issue, I ended up adding a one to many dependency and using projection I was able to fetch the data the way i wanted. Pretty powerful on EF Core's end that an FK is not actually required so I can have a "fake" mapping between two views even though it's not a real FK. Good stuff!!!
